packages/http/src/getMockExtensions.ts

```diff
@@ -0,0 +1,153 @@
+import { existsSync, readdirSync } from 'fs';
+import { defaults, defaultsDeep, pick } from 'lodash';
+import { basename, extname, join, resolve } from 'path';
+
+import {
+  IJsonSchemaFakerExtensionConfig,
+  IJsonSchemaFakerExtensionDirectories,
+  IJsonSchemaFakerExtensionDirectoryNames,
+  IJsonSchemaFakerExtensions,
+  IJsonSchemaFakerExtensionTypeKey,
+  IJsonSchemaFakerExtensionTypes,
+  JsonSchemaFakerCustomFormatExtension,
+  JsonSchemaFakerCustomGeneratorExtension,
+  JsonSchemaFakerExtension,
+  JsonSchemaFakerExtensionValidator,
+  JsonSchemaFakerExternalGeneratorExtension,
+  RecursivePartial,
+} from './types';
+
+export const defaultDirectories: IJsonSchemaFakerExtensionDirectories = {
+  baseDirectory: `${process.cwd()}/prism`,
+  customFormatsDirectory: 'custom-formats',
+  externalGeneratorsDirectory: 'external-generators',
+  customGeneratorsDirectory: 'custom-generators',
+};
+
+export const defaultExtensions: IJsonSchemaFakerExtensionTypes = {
+  customFormats: {},
+  externalGenerators: {},
+  customGenerators: {},
+};
+
+export const defaultConfig: IJsonSchemaFakerExtensionConfig = {
+  watchDirectories: true,
+  directories: defaultDirectories,
+  ...defaultExtensions,
+};
+
+// Check the extension type against the given operands
+const typesValidator = (extension: JsonSchemaFakerExtension, types: string[]) =>
+  types.some(type => {
+    if (typeof extension === 'object') {
+      return Object.keys(extension).length > 0;
+    } else {
+      return typeof extension === type;
+    }
+  });
+
+const customFormatExtensionValidator = (
+  extension: JsonSchemaFakerExtension,
+): extension is JsonSchemaFakerCustomFormatExtension => typesValidator(extension, ['number', 'string', 'function']);
+
+const externalGeneratorExtensionValidator = (
+  extension: JsonSchemaFakerExtension,
+): extension is JsonSchemaFakerExternalGeneratorExtension => typesValidator(extension, ['function']);
+
+const customGeneratorExtensionValidator = (
+  extension: JsonSchemaFakerExtension,
+): extension is JsonSchemaFakerCustomGeneratorExtension => typesValidator(extension, ['number', 'string', 'object']);
+
+const validateExtensions = <T>(extensions: T, validator: JsonSchemaFakerExtensionValidator): T =>
+  Object.keys(extensions).reduce((acc, extensionKey) => {
+    const extension = extensions[extensionKey];
+    if (validator(extension)) {
+      return {
+        ...acc,
+        [extensionKey]: extension,
+      };
+    } else {
+      // TODO: Use Prism logging to warn use about the invalid module
```

Comment:
@philsturgeon 

I'd like to replace this warning with a message outputted by the Logger Prism uses, only, I'm far from experienced with the `fp-ts` paradigm you are using in this project. Some help here would be much appreciated.